pkg/proxy/balancer/weight.go

```diff
@@ -0,0 +1,50 @@
+package balancer
+
+import (
+	"errors"
+	"math/rand"
+)
+
+type (
+	// WeightBalancer balancer
+	WeightBalancer struct{}
+)
+
+var (
+	// ErrZeroWeight is used when there a zero value weight was given
+	ErrZeroWeight = errors.New("invalid backend, weight 0 given")
+)
+
+// NewWeightBalancer creates a new instance of Roundrobin
+func NewWeightBalancer() *WeightBalancer {
+	return &WeightBalancer{}
+}
+
+// Elect backend using roundrobin strategy
+func (b *WeightBalancer) Elect(hosts []*Target) (*Target, error) {
+	if len(hosts) == 0 {
+		return nil, ErrEmptyBackendList
+	}
+
+	totalWeight := 0
+	for _, host := range hosts {
+		totalWeight += host.Weight
+	}
+
+	if totalWeight <= 0 {
+		return nil, ErrZeroWeight
+	}
+
+	r := rand.Intn(totalWeight)
```

Comment:
I think we should add something like this to init rand seed

```
func init() {
	rand.Seed(time.Now().UnixNano())
} 
```

Reply:
It's being initialized on the balancer.go file. It's for the whole package